test/integration/targets/firewalld/tasks/main.yml

```diff
@@ -0,0 +1,28 @@
+# Test playbook for the firewalld module
+# (c) 2017, Adam Miller <admiller@redhat.com>
+
+# This file is part of Ansible
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+
+
+# Have to conditionalize this based on the version of python used as a dep to
+# firewalld by distros
+#
+# python2 - RHEL 7+, CentOS 7+, SLES 11/12, Ubuntu 14.04 LTS, openSUSE Leap 42.2
+# python3 - Fedora 24+, debian 8, and Ubuntu 16.04 LTS+
+- import_tasks: test_cases.yml
+  when: (ansible_distribution in ["Fedora", "Ubuntu", "Debian", "Arch"] and ansible_python["version"]["major"] == 3) or
+        (ansible_distribution in ["SUSE", "openSUSE Leap"] and ansible_python["version"]["major"] == 2) or
+        (ansible_distribution in ["RedHat", "CentOS"] and ansible_distribution_major_version|int >= 7)
```

Comment:
Is there a reasonable way to detect the python version used instead of hard-coding the distributions and python versions into the test?

Here is what I see running in CI:

- rhel/7.4
- linux/centos7
- linux/opensuse42.2
- linux/opensuse42.3
- linux/ubuntu1604py3

Reply:
+1 will do, thanks

Reply:
@mattclay so I went to fix this up and I remembered the issue isn't what version of python, but what version of python on which distro because some distros only have python2 or python3 versions of firewalld available. Therefore if the tests detect that the `ansible_python_interpreter` is python3, that's not enough information, we unfortunately also have to know what distro/release it's on to know if the test is good to go.

Thoughts? I'm completely open to a better solution, I agree this is ugly.

Reply:
Is it sufficient to perform an `import firewall.config` in a try/except block with the current python interpreter to determine if the test can run?

Reply:
@mattclay yup, that'll work. :+1: